go-controller/pkg/metrics/metrics.go

```diff
@@ -43,12 +44,39 @@ type metricDetails struct {
 	metric        prometheus.Gauge
 }
 
+type stopwatchMetricDetails struct {
+	srcName string
+	metrics struct {
+		totalSamples   prometheus.Gauge
```

Comment:
`totalSamples` seems to be a cumulative variable (that's the impression I get from the test cases you wrote), we should be using `prometheus.Counter` if I read the docs right https://prometheus.io/docs/concepts/metric_types/#counter ?